vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1295,6 +1298,33 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
+    private boolean hasSiblingWithSameType(Node child) throws RepositoryException {
+
+        Node parent = child.getParent();
+
+        try {
+            EffectiveNodeType ent = EffectiveNodeType.ofNode(parent);
+            String typeName = ent.getApplicableChildNodeDefinition(child.getName(), child.getPrimaryNodeType()).get().getName();
```

Comment:
You should consider the mixins as well!

Reply:
I the EffectiveNodeType already computes the, well, effective type, which is a merge of all the super and mixin types.

Reply:
I am not talking about line 6 which takes into account mixins and inheritance, but about line 7 which retrieves the applicable mixin. IMHO this needs to parameterized with name, primary and mixin types otherwise the wrong applicable child node definition might be returned (the child node might actually be enforced to have a certain mixin instead of a primary)

Reply:
Again: how is this relevant here?

Reply:
Example: 

1. Parent enforces mandatory child of mixin type m1
2. Parent contains two childs of that mixin 
3. `ent.getApplicableChildNodeDefinition(child.getName(), child.getPrimaryNodeType()`) will return `Optional.empty()` in that case
3. That leads to `hasSiblingWithSameType()` returning `false` which is obviously wrong here

Reply:
but `ent.getApplicableChildNodeDefinition()` already operates on the merged node type....

Reply:
Seems that actually 1. is not supported by JCR: https://www.adobe.io/experience-manager/reference-materials/spec/jcr/2.0/3_Repository_Model.html#3.7.4.1%20Required%20Primary%20Node%20Types: Only primary types can be enforced but not mixins.